api_server.py

```
@@ -160,9 +164,11 @@ def check_rules(available_ids: List[int]) -> Dict:
         }
     with get_db() as conn:
         placeholders = ",".join("?" * len(available_ids))
-        rows = conn.execute(
-            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})", available_ids
-        ).fetchall()
+        # Building the placeholders string is safe here as it only contains '?' characters.
+        query = (
+            f"SELECT role, skills FROM crew WHERE id IN ({placeholders})"  # nosec B608
```

Ruff (0.15.0)

[error] Possible SQL injection vector through string-based query construction

(S608)